cylc/flow/data_store_mgr.py

```diff
@@ -2521,19 +2535,72 @@ def delta_task_ext_trigger(
         ext_trigger.time = update_time
         self.updates_pending = True
 
+    @staticmethod
+    def _set_task_xtrigger_modifiers(node_or_delta):
+        # update the xtrigger task modifiers
+        node_or_delta.is_retry = False
+        node_or_delta.is_wallclock = False
+        node_or_delta.is_xtriggered = False
+        for xtrigger in node_or_delta.xtriggers.values():
+            if xtrigger.satisfied:
+                continue
+            if (
+                xtrigger.label.startswith('_cylc_retry')
+                or xtrigger.label.startswith('_cylc_submit_retry')
+            ):
+                node_or_delta.is_retry = True
+            elif xtrigger.id.startswith('wall_clock'):
+                node_or_delta.is_wallclock = True
+            else:
+                node_or_delta.is_xtriggered = True
+
     def _delta_xtrigger(
-        self, tp_id: str, label: str, sig: str, satisfied: bool,
-        t_update: float
+        self,
+        tp_id: str,
+        label: str,
+        sig: str,
+        satisfied: bool,
+        update_time: float,
     ) -> None:
         """Helper for the two xtrigger delta methods."""
+        # fetch the task from the store
+        tp_id, tproxy = self.store_node_fetcher(Tokens(tp_id))
+        if not tproxy:
+            return
+
+        # create or fetch the updated delta
         tp_delta = self.updated[TASK_PROXIES].setdefault(
-            tp_id, PbTaskProxy(id=tp_id))
-        tp_delta.stamp = f'{tp_id}@{t_update}'
+            tp_id, PbTaskProxy(id=tp_id)
+        )
+        tp_delta.stamp = f'{tp_id}@{update_time}'
+
+        # populate all xtriggers on the delta if not already present
+        if not tp_delta.xtriggers:
+            # NOTE: if one xtrigger changes, we must include all in the
+            # delta, see https://github.com/cylc/cylc-flow/issues/6307
+            for _sig, xtrigger in tproxy.xtriggers.items():
+                if _sig == sig:
+                    # don't copy the xtrigger we are changing
+                    continue
+                _xtrigger = tp_delta.xtriggers[_sig]
+                _xtrigger.id = xtrigger.id
+                _xtrigger.label = xtrigger.label
+                _xtrigger.satisfied = xtrigger.satisfied
+                _xtrigger.time = xtrigger.time
```

Comment:
This part fixes the incremental update issue - https://github.com/cylc/cylc-flow/pull/6671/commits/14f9266cce16cce9b484af951fe30af81c73a3d5****